advocacy_docs/partner_docs/HashicorpVault/05-UsingHashicorpVault.mdx

```diff
@@ -0,0 +1,241 @@
+- Host
+- Port
+- Username
+- Password
+- Keyfile
+- Certfile
+- Ca_certs
+
+In the below example, Username and Password are commented out as Hashicorp Vault does not utilize those for their connectivity:
+```bash
+ubuntu@ip-172-31-46-134:/tmp$ cat pykmip.conf 
+[client]
+host=127.0.0.1
+port=5696
+#username=     
+#password=
+keyfile=/tmp/key.pem
+certfile=/tmp/cert.pem
+ca_certs=/tmp/ca.pem
+#cert_reqs=CERT_REQUIRED
+```
+
+!!! Note
+    For more information on the pykmip.conf file and the contents of it you can visit the [pykmip documentation](https://pykmip.readthedocs.io/en/latest/client.html).
+
+## Create a Key on Hashicorp Vault Secrets Engine
+
+1. On your system where you have your EDB Postgres Advanced Server 15.2 instance, assume root user to create the key on the Hashicorp Vault Secrets Engine.
+
+2. Type `python3` and then input the following, making adjustments per your system setup and directory paths:
+
+```bash
+>>> from kmip.pie import client
+>>> from kmip import enums
+>>> c = client.ProxyKmipClient(config_file='/tmp/pykmip.conf')
+>>> c.open()
+>>> key_id = c.create(enums.CryptographicAlgorithm.AES, 128, name='edbtestkey')
+>>> c.activate(key_id)
+>>> key_id
+>>> 'key_output_shows_here'
+>>> c.close()
+
+```
+3. If this runs without error then your key has been successfully created.  You cannot view keys that you create in Hashicorp Vault.
+
+## Verify Encryption and Decryption
+
+To ensure that your key that you created will be able to encrypt and decrypt data, run the following two commands as the root user on your system where you have your EDB Postgres Advanced Server 15.2 instance.
+
+1. `printf secret | python3 /tmp/edb_tde_kmip_client.py encrypt --out-file=test.bin --pykmip-config-file=/tmp/pykmip.conf  --key-uid='key_output_here’ --variant=pykmip`
+
+2. `python3 /tmp/edb_tde_kmip_client.py decrypt --in-file=test.bin --pykmip-config-file=/tmp/pykmip.conf --key-uid='key_output_here' --variant=pykmip`
+
+3. If this is successful it should produce the output of **secret**.
+
+```bash
+root@ip-172-31-46-134:/etc/vault.d# printf secret | python3 /tmp/edb_tde_kmip_client.py encrypt --out-file=test.bin --pykmip-config-file=/tmp/pykmip.conf --key-uid='nfTCV2Cp5sffhQuRrOVfgCUyu8qh9kwd' --variant=pykmip
+root@ip-172-31-46-134:/etc/vault.d# python3 /tmp/edb_tde_kmip_client.py decrypt --in-file=test.bin --pykmip-config-file=/tmp/pykmip.conf --key-uid='nfTCV2Cp5sffhQuRrOVfgCUyu8qh9kwd' --variant=pykmip
+Secret
+root@ip-172-31-46-134:/etc/vault.d# 
+```
+
+## Perform initdb for the Database
+
+After you have completed the above steps you will be able to export the PGDATAKEYWRAPCMD and PGDATAKEYUNWRAPCMD to wrap and unwrap your encryption key and initialize your database.
+
+1. Login to your EDB Postgres Advanced Server system.
+
+2. Navigate to the `/bin` directory where your executables live, for example `/usr/lib/edb-as/15/bin` would be the executables directory for an Ubuntu system.
```

Comment:
I don't think we need to say for an Ubuntu system. 
We can say 
In our example it is /usr/lib/edb-as/15/bin